### PR TITLE
[backup] the EpochEnding backup type.

### PR DESCRIPTION
**Pull request by msmouse**
## Motivation

It also serves as the basis of proof for all other types of backups.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan
unit tests

### CODE REVIEW

File: execution/executor/tests/db_bootstrapper_test.rs [hunk 1]
```diff
@@ -239,8 +239,12 @@ fn restore_state_to_db(
     root_hash: HashValue,
     version: Version,
 ) {
-    db.restore_account_state(vec![(accounts, proof)].into_iter(), version, root_hash)
-        .unwrap();
+    let rh = db.get_restore_handler();
+    let mut restore = rh.get_state_restore_receiver(version, root_hash).unwrap();
```
Comment (wqfish) [inline execution/executor/tests/db_bootstrapper_test.rs:243]: maybe name this `receiver`?

File: storage/backup/backup-cli/src/backup_types/state_snapshot/tests.rs [hunk 2]
```diff
@@ -65,4 +65,6 @@ fn end_to_end() {
             .account_state_root_hash,
         state_root_hash,
     );
+
+    drop(rt);
```
Comment (wqfish) [inline storage/backup/backup-cli/src/backup_types/state_snapshot/tests.rs:69]: i see this in a few places. what does it do?

Comment (msmouse) [reply storage/backup/backup-cli/src/backup_types/state_snapshot/tests.rs:69]: it's a mistake, I should do `rt.shutdown_timeout()`. updating.

**Pull request closed without merging**